src/main/ruby/truffleruby/core/kernel.rb

```diff
@@ -726,17 +726,23 @@ def printf(*args)
   end
 
   def caller(start = 1, limit = nil)
-    args = if start.is_a? Range
-             if Primitive.nil? start.end
-               [start.begin + 1]
-             else
-               [start.begin + 1, start.size]
-             end
-           elsif Primitive.nil? limit
-             [start + 1]
-           else
-             [start + 1, limit]
-           end
+    args =  if start.is_a? Range
+              if (Primitive.nil? start.begin) and (Primitive.nil? start.end)
```

Comment:
Could you put the parens around the arguments like `Primitive.nil?(start.begin)`?
I think that's a lot more common, and parens around a condition are rare in Ruby, unless they need to affect grouping of conditions like (a || b) && c.